streams/src/test/java/org/apache/kafka/streams/integration/ConsistencyVectorIntegrationTest.java

```diff
@@ -131,67 +129,56 @@ public void shouldHaveSamePositionBoundActiveAndStandBy() throws Exception {
-                assertThat(store.getPosition().getPartitionPositions(INPUT_TOPIC_NAME), hasEntry(0, 99L));
-                count.incrementAndGet();
-            }
-        }
-        assertThat(count.get(), is(2));
-    }
-
-    public class TestingRocksDBStore extends RocksDBStore {
-        public TestingRocksDBStore(final String name, final String metricsScope) {
-            super(name, metricsScope);
-        }
-
-        public Position getPosition() {
-            return position;
-        }
-
-        public File getDbDir() {
-            return dbDir;
-        }
+        final StateQueryRequest<ValueAndTimestamp<Integer>> request =
+            StateQueryRequest
+                .inStore(TABLE_NAME)
+                .withQuery(KeyQuery.<Integer, ValueAndTimestamp<Integer>>withKey(key))
+                .withPositionBound(PositionBound.unbounded());
+
+        checkPosition(batch1NumMessages, request, kafkaStreams1);
+        checkPosition(batch1NumMessages, request, kafkaStreams2);
     }
 
-    public class TestingRocksDbKeyValueBytesStoreSupplier implements KeyValueBytesStoreSupplier {
-
-        public List<TestingRocksDBStore> stores = new LinkedList<>();
-
-        private final String name;
-
-        public TestingRocksDbKeyValueBytesStoreSupplier(final String name) {
-            this.name = name;
-        }
-
-        @Override
-        public String name() {
-            return name;
-        }
+    private void checkPosition(final int batch1NumMessages,
+                               final StateQueryRequest<ValueAndTimestamp<Integer>> request,
+                               final KafkaStreams kafkaStreams1) throws InterruptedException {
+        final long maxWaitMs = TestUtils.DEFAULT_MAX_WAIT_MS;
+        final long expectedEnd = System.currentTimeMillis() + maxWaitMs;
+
+        while (true) {
+            final StateQueryResult<ValueAndTimestamp<Integer>> stateQueryResult =
+                IntegrationTestUtils.iqv2WaitForResult(
+                    kafkaStreams1,
+                    request
+                );
+            final QueryResult<ValueAndTimestamp<Integer>> queryResult =
+                stateQueryResult.getPartitionResults().get(0);
+            if (queryResult.isSuccess() && queryResult.getResult() != null) {
+                // invariant: each value is also at the equivalent offset
+                assertThat(
+                    "Result:" + queryResult,
+                    queryResult.getPosition(),
+                    is(
+                        Position.emptyPosition()
+                                .withComponent(INPUT_TOPIC_NAME, 0, queryResult.getResult().value())
+                    )
+                );
+
+                if (queryResult.getResult().value() == batch1NumMessages - 1) {
+                    // we're at the end of the input.
+                    return;
+                }
```

Comment:
```suggestion
                // we're at the end of the input.
                if (queryResult.getResult().value() == batch1NumMessages - 1) return;
```

Reply:
Thanks for the review, @YeonCheolGit !

I know that some of the core code looks different, but in the Streams module, we try to always use control flow brackets.